org.adempiere.base/src/org/compiere/model/MPayment.java

```diff
@@ -1938,7 +1938,8 @@ public String prepareIt()
 				if (length > 0)		//	get last invoice
 					setC_Invoice_ID (invoices[length-1].getC_Invoice_ID());
 				//
-				if (getC_Invoice_ID() == 0)
+				MDocType orderDocType = new MDocType(getCtx(), order.getC_DocType_ID(), get_TrxName());

```

Comment:
for better performance, should use `MDocType.get(order.getC_DocType_ID())` instead